### ISSUE TITLE
babel-register doesn't resolve paths as expected

### ISSUE THREAD

**Issue by myuseringithub** [babel/babel#8453]
When loading `babel-register` from a different location than the Node app's entrypoint (require.main.filename`), it doesn't resolve the babel plugin/preset modules as expected: 
1. `NODE_PATH` environment variable is ignored and not being used by `babel-register` to load the passed configuration plugins/presets.
2. Module resolution lookup doesn't start from the loaded path of `babel-register`, instead the babel require hook module resolution begins from the entrypoint file path. 

_Possible solution:_  Add an option to be passed to `babel-register` function that will extend the resolution paths when first loading the plugins/presets. 

_Version:_ Babel 7, Node >10, Linux container

**Comment by loganfsmyth** [comment]
1) Generally the expectation is that plugins will be installed relative to the location they are referenced, since otherwise you're separating responsibility for compilation and versioning which is likely to cause issues. 
2) Plugin/presets are resolved either relative to the config file that references the plugin, or the working directory. I'm not sure I understand what you mean `the loaded path of babel-register`, or why that would be desirable in this context.

It would help if you could elaborate on the actual directory structure you have and why you feel the current behavior is insufficient.

**Comment by myuseringithub** [comment]
Well, an App repository with Git submodule for handling all that's related to transpilation - https://github.com/AppScriptIO/javascriptTranspilation/blob/master/entrypoint.js#L26
 
App project: specifies pacakges used in the source code.
•  Git submodule: javascript transpilation: specifies packages related to Babel.

**Comment by loganfsmyth** [comment]
Generally in contexts like that, my expectation is that you'd replace lines like https://github.com/AppScriptIO/javascriptTranspilation/blob/ae2490c6ed0b40f158c34a3f6da07879fb9c4641/compilerConfiguration/serverBuild.BabelConfig.js#L13 with

```
require("@babel/plugin-syntax-dynamic-import")
```
and not have `@babel/register` have to resolve the string at all.

**Comment by myuseringithub** [comment]
Can I pass the plugin function instead of a string in the options ? Well it will probably solve my issue.

**Comment by loganfsmyth** [comment]
Yeah, you can pass the loaded plugin directly if you like. Also I'll say that it's also possible to replace
```
path.normalize(`${__dirname}/../node_modules/@babel/plugin-syntax-dynamic-import`),
```
```
require.resolve(`@babel/plugin-syntax-dynamic-import`),
```
to avoid having to hardcode `__dirname` and `..` and `node_modules` and such.

**Comment by myuseringithub** [comment]
@loganfsmyth I think it should be mentioned somewhere in the docs that babel configuration accepts loaded plugins not only strings.
